pkgs/dart_tooling_mcp_server/test/tools/dtd_test.dart

```diff
@@ -25,65 +27,102 @@ void main() {
+        test('can take a screenshot', () async {
+          await testHarness.startDebugSession(
+            counterAppPath,
+            'lib/main.dart',
+            isFlutter: true,
+          );
+          final tools =
+              (await testHarness.mcpServerConnection.listTools()).tools;
+          final screenshotTool = tools.singleWhere(
+            (t) => t.name == DartToolingDaemonSupport.screenshotTool.name,
+          );
+          final screenshotResult = await testHarness.callToolWithRetry(
+            CallToolRequest(name: screenshotTool.name),
+          );
+          expect(screenshotResult.content.single, {
+            'data': anything,
+            'mimeType': 'image/png',
+            'type': ImageContent.expectedType,
+          });
         });
-      });
 
-      test('can perform a hot reload', () async {
-        await testHarness.startDebugSession(
-          counterAppPath,
-          'lib/main.dart',
-          isFlutter: true,
-        );
-        final tools = (await testHarness.mcpServerConnection.listTools()).tools;
-        final hotReloadTool = tools.singleWhere(
-          (t) => t.name == DartToolingDaemonSupport.hotReloadTool.name,
-        );
-        final hotReloadResult = await testHarness.callToolWithRetry(
-          CallToolRequest(name: hotReloadTool.name),
-        );
+        test('can get the widget tree', () async {
+          await testHarness.startDebugSession(
+            counterAppPath,
+            'lib/main.dart',
+            isFlutter: true,
+          );
+          final tools =
+              (await testHarness.mcpServerConnection.listTools()).tools;
+          final getWidgetTreeTool = tools.singleWhere(
+            (t) => t.name == DartToolingDaemonSupport.getWidgetTreeTool.name,
+          );
+          final getWidgetTreeResult = await testHarness.callToolWithRetry(
+            CallToolRequest(name: getWidgetTreeTool.name),
+          );
 
-        expect(hotReloadResult.isError, isNot(true));
-        expect(hotReloadResult.content, [
-          TextContent(text: 'Hot reload succeeded.'),
-        ]);
+          expect(getWidgetTreeResult.isError, isNot(true));
+          expect(
+            (getWidgetTreeResult.content.first as TextContent).text,
+            contains('MyHomePage'),
+          );
+        });
       });
 
-      test('can get the widget tree', () async {
-        await testHarness.startDebugSession(
-          counterAppPath,
-          'lib/main.dart',
-          isFlutter: true,
-        );
-        final tools = (await testHarness.mcpServerConnection.listTools()).tools;
-        final getWidgetTreeTool = tools.singleWhere(
-          (t) => t.name == DartToolingDaemonSupport.getWidgetTreeTool.name,
-        );
-        final getWidgetTreeResult = await testHarness.callToolWithRetry(
-          CallToolRequest(name: getWidgetTreeTool.name),
-        );
+      group('dart cli tests', () {
+        test('can perform a hot reload', () async {
```

Comment:
seems like we should still have testing that we can hot reload a flutter app since this is a different code path than hot reloading a dart app (detecting the hot reload service name)

Reply:
I need to separately validate that the custom flutter logic is actually necessary as well.

But I do generally agree, I am still not sure how we are going to manage these flutter tests exactly though.

Reply:
I will add it back for now, we have other ones anyways